src/clip.sol

```diff
@@ -272,6 +274,7 @@ contract Clipper {
         if (owe >= sale.tab) {
             owe = sale.tab;       // owe' <= owe
             slice = owe / price;  // Adjust slice; slice' = owe' / price <= owe / price == slice <= lot
+            // TODO check that this doesn't violate max from user
```

Comment:
It doesn't since `owe'` is now lower or equal than its original value:

    slice' = owe' / price // from line 275 we know that owe' = sale.tab
    slice' = sale.tab / price // from line 274 we know that owe >= sale.tab
    slice' <= owe / price // from line 269 we know that owe = slice * price
    slice' <= (slice * price) / price
    slice' <= slice

Reply:
Yeah it shouldn't violate the max price as `slice` is actually defined as `owe / price` as Emilio said.
Btw this is inside the function we might restructure.